### PR TITLE
Data Entity's overview test report shows stats of test runs instead of tests

### LINKED ISSUE #616: Refactor DataQualityTests to the reactive approach

### LINKED ISSUE #711: Data Entity's overview test report shows stats of test runs instead of tests

### LINKED ISSUE #717: Implement a stub in the test report score

### PR DESCRIPTION
**Pull request by DementevNikita**
Closes #711 
Closes #717 
Closes #616

### CODE REVIEW

File: odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/service/DataQualityServiceImpl.java [hunk 1]
```diff
@@ -8,39 +9,44 @@
...
+    private final DataEntityRepository dataEntityRepository;
     private final DataQualityMapper dataQualityMapper;
     private final DataEntityMapper dataEntityMapper;
 
     @Override
-    public Mono<DataEntityList> getDataEntityDataQATests(final long dataEntityId) {
-        return Mono.fromCallable(() -> dataQualityRepository.getDataQualityTests(dataEntityId))
+    public Mono<DataEntityList> getDatasetTests(final long datasetId) {
+        return dataQualityRepository.getDataQualityTestOddrnsForDataset(datasetId)
+            .switchIfEmpty(Mono.error(new NotFoundException("Dataset with id %d not found".formatted(datasetId))))
```
Comment (damirabdul) [inline odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/service/DataQualityServiceImpl.java:28]: "Data quality tests for dataset with id %d not found"

File: odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/repository/reactive/ReactiveDataQualityRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            .flatMap(list -> jooqQueryHelper.pageifyResult(
+                list,
+                r -> r.into(DataEntityTaskRunPojo.class),
+                fetchCount(baseQuery)
+            ));
+    }
+
+    @Override
+    public Mono<DatasetTestReportDto> getDatasetTestReport(final long datasetId) {
+        final SelectConditionStep<Record1<LocalDateTime>> maxEndTimeSubquery = DSL
```
[the remark this answers is not in the document]
Comment (DementevNikita) [reply odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/repository/reactive/ReactiveDataQualityRepositoryImpl.java:91]: We need to take the latest run, so I'm filtering by the max(end_time) in the outer query

File: odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/service/DataQualityServiceImpl.java (hunk 1 above)
Comment (DementevNikita) [reply odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/service/DataQualityServiceImpl.java]: Fixed

File: odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/repository/reactive/ReactiveDataQualityRepositoryImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (DementevNikita) [reply odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/repository/reactive/ReactiveDataQualityRepositoryImpl.java:91]: discussed offline, added an index to end_time column

**Review by damirabdul — APPROVED** [review]

**Pull request merged**